src/DotVVM.Framework/Controls/DataPager.cs

```diff
@@ -138,12 +138,12 @@ public bool Enabled
         public static readonly DotvvmProperty EnabledProperty =
             DotvvmPropertyWithFallback.Register<bool, DataPager>(nameof(Enabled), FormControls.EnabledProperty);
 
-        private HtmlGenericControl content;
-        private HtmlGenericControl firstLi;
-        private HtmlGenericControl previousLi;
-        private PlaceHolder numbersPlaceHolder;
-        private HtmlGenericControl nextLi;
-        private HtmlGenericControl lastLi;
+        protected HtmlGenericControl content;
+        protected HtmlGenericControl firstLi;
+        protected HtmlGenericControl previousLi;
+        protected PlaceHolder numbersPlaceHolder;
+        protected HtmlGenericControl nextLi;
+        protected HtmlGenericControl lastLi;
```

Comment:
I think I don't like this. Can we instead have protected virtual methods like `CreateFirstLi()`?

Reply:
I'm not sure if methods would be proper here. Maybe protected property with more generic name would be sufficient?

Reply:
At least properties, but not fields.

Reply:
@quigamdev Is there any difference between properties and fields? I know that you could potentially reimplement the property setter and getter to maintain binary compatibility, but if we'd remove these fields it would not make any sense to emulate them somehow, as these controls would not exist anymore and thus manipulating them whould be useless.